src/hidimstat/knockoffs.py

```diff
@@ -1,109 +1,570 @@
+            LogisticRegressionCV(penalty="l1", max_iter=1000, solver="liblinear",
+                cv=KFold(n_splits=5, shuffle=True, random_state=0), n_jobs=None, tol=1e-8)
+            LogisticRegressionCV(penalty="l2", max_iter=1000, n_jobs=None,
+                verbose=0, cv=KFold(n_splits=5, shuffle=True, random_state=0), tol=1e-8,)
 
-    offset : int, 0 or 1, optional
-        offset to calculate knockoff threshold, offset = 1 is equivalent to
-        knockoff+
+    preconfigure_estimator : function, optional
+        A function that configures the estimator for the Model-X knockoff procedure.
+        If provided, this function will be called with the estimator, X, X_tilde, and y
+        as arguments, and should modify the estimator in-place.
 
-    method : str, optional
-        knockoff construction methods, either equi for equi-correlated knockoff
-        or sdp for optimization scheme
+    centered : bool, optional (default=True)
+        Whether to standardize the data before performing the inference procedure.
 
-    statistics : str, optional
-        method to calculate knockoff test score
+    cov_estimator : sklearn covariance estimator instance, optional
+        The method used to estimate the empirical covariance matrix. This can be any
+        estimator with a `fit` method that accepts a 2D array and a `covariance_`
+        attribute that returns a 2D array of the estimated covariance matrix.
+        Examples include LedoitWolf and GraphicalLassoCV.
 
-    shrink : bool, optional
-        whether to shrink the empirical covariance matrix
+    joblib_verbose : int, optional (default=0)
+        The verbosity level of the joblib parallel processing.
 
-    centered : bool, optional
-        whether to standardize the data before doing the inference procedure
+    n_bootstraps : int, optional (default=1)
+        The number of bootstrap samples to generate for aggregation.
 
-    cov_estimator : str, optional
-        method of empirical covariance matrix estimation
+    n_jobs : int, optional (default=1)
+        The number of jobs to run in parallel.
 
-    seed : int or None, optional
-        random seed used to generate Gaussian knockoff variable
+    random_state : int or None, optional (default=None)
+        The random seed used to generate the Gaussian knockoff variables.
 
-    Returns
-    -------
-    selected : 1D array, int
-        vector of index of selected variables
+    tol_gauss : float, optional (default=1e-14)
+        A tolerance value used for numerical stability in the calculation of the Cholesky decomposition in the gaussian generation function.
 
-    test_score : 1D array, (n_features, )
-        vector of test statistic
+    fdr : float, optional (default=0.1)
+        The desired controlled False Discovery Rate (FDR) level.
 
-    thres : float
-        knockoff threshold
+    offset : int, 0 or 1, optional (default=1)
```

Comment:
The offset should never be other than 1. This will have to be removed in a forthcoming PR.